packages/metro/src/ModuleGraph/worker/collectDependencies.js

```diff
@@ -39,18 +39,20 @@ export type Dependency<TSplitCondition> = $ReadOnly<{
 }>;
 
 // TODO: Convert to a Flow enum
-type ContextMode = 'sync' | 'eager' | 'lazy' | 'lazy-once';
+export type ContextMode = 'sync' | 'eager' | 'lazy' | 'lazy-once';
 
-type ContextFilter = {pattern: string, flags: string};
+export type ContextFilter = {pattern: string, flags: string};
 
-export type RequireContextParams = $ReadOnly<{
+export type RequireContextParams = {
+  /** Absolute file path pointing to the root directory of the context. */
+  from?: string,
```

Comment:
We should split this data structure - `collectDependencies` will *never* populate `from` with an absolute path because those aren't cacheable, whereas later in the pipeline we need `from` to *always* be present.